src/dotnet/APIView/APIViewWeb/Managers/CopilotCommentsManager.cs

```diff
@@ -31,163 +31,136 @@ IConfiguration configuration
                 new AzureKeyCredential(_configuration["OpenAI:Key"]));
         }
 
-        public async Task<string> CreateDocumentAsync(string user, string badCode, string goodCode, string language, string comment, string[] guidelineIds)
+        public async Task<string> CreateDocumentAsync(string language, string badCode, string goodCode, string comment, string guidelineIds, string user)
         {
-            var embedding = await GetEmbeddingsAsync(badCode);
-
             var document = new CopilotCommentModel()
             {
+                Language = language,
                 BadCode = badCode,
                 GoodCode = goodCode,
-                Embedding = embedding,
-                Language = language,
+                Embedding = await GetEmbeddingsAsync(badCode),
```

Comment:
I don't know what .NET returns here - is this just the embedding data or does it return an object with metadata and the embedding?

Reply:
`GetEmbeddingsAsync` is a function I defined to extract just the vector values from an `Embeddings` object which contains metadata and the vector. We don't need the metadata since we're just computing the cosine similarity.